cohortextractor/main.py

```diff
@@ -29,30 +35,37 @@ def generate_cohort(
         write_output(results, output_file)
 
 
-def generate_measures(definition_path, output_file):
-    measures = load_measures(definition_path)
+def generate_measures(definition_path, input_file, output_file):
+    cohort = load_cohort(definition_path)
     output_file.parent.mkdir(parents=True, exist_ok=True)
-    for measure in measures:
-        measure_output_file = str(output_file).replace("*", measure.id)
-        results = measure.calculate()
-        results.to_csv(measure_output_file)
+
+    for measure_id, results in calculate_measures_results(cohort, input_file):
+        measure_output_file = str(output_file).replace("*", measure_id)
+        results.to_csv(measure_output_file, index=False)
+        log.info("Created measure output", output=output_file)
+
+
+def calculate_measures_results(cohort, input_file, patient_dataframe=None):
+    measures = get_measures(cohort)
+    measures_manager = MeasuresManager(measures, input_file, patient_dataframe)
+    assert (
+        measures_manager.patient_dataframe is not None
+    ), "No matching input file found. You may need to first run:\n  cohortextractor generate_cohort ..."
```

Comment:
It seems a shame to expose this `patient_dataframe` property just to do the assertion, when the rest of the interface to that object is so simple. Could we do the assertion inside `MeasureManager` instead?

Reply:
Yes - in fact it more or less does already, I'll move it over

Reply:
Although I vaguely think that the MeasuresManager only takes a `patient_dataframe` parameter in order to allow it to be passed in tests; I'll have another look at that too